cl/phase1/stages/stage_history_download.go

```diff
@@ -188,6 +188,7 @@ func SpawnStageHistoryDownload(cfg StageHistoryReconstructionCfg, ctx context.Co
 	}()
 
 	go func() {
+		ctx := context.Background()
```

Comment:
hmmm, this ctx won't stop.
remove this and use input parameter ctx in line 72 instead?